docs/traffic-control.md

```diff
@@ -0,0 +1,277 @@
+from specific Pods or destined to specific Pods to a local network device or a
+remote destination via various tunnels. It provides full visibility into network
+traffic, including both north-south and east-west traffic.
+
+You may be interested in using this capability if any of the following apply:
+
+- You want to monitor network traffic passing in or out of a set of Pods for
+  purposes such as troubleshooting, intrusion detection, and so on.
+
+- You want to redirect network traffic passing in or out of a set of Pods to
+  applications that enforce policies, and reject traffic to prevent intrusion.
+
+This guide demonstrates how to configure `TrafficControl` to achieve the above
+goals.
+
+## Prerequisites
+
+TrafficControl was introduced in v1.7 as an alpha feature. A feature gate,
+`TrafficControl` must be enabled on the antrea-agent in the `antrea-config`
+ConfigMap like the following options for the feature to work:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    featureGates:
+      TrafficControl: true
+```
+
+## The TrafficControl resource
+
+A TrafficControl in Kubernetes is a REST object. Like all the REST objects, you
+can POST a TrafficControl definition to the API server to create a new instance.
+For example, suppose you have a set of Pods which contain a label `app=web`, the
+following specification creates a new TrafficControl object named
+"mirror-web-app", which mirrors all traffic from or to any Pod with the
+`app=web` label and send them to a receiver running on "10.0.10.2" encapsulated
+within a VXLAN tunnel:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: mirror-web-app
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        app: web
+  direction: Both
+  action: Mirror
+  targetPort:
+    vxlan:
+      remoteIP: 10.0.10.2
+```
+
+### AppliedTo
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+TrafficControl applies to. Pods can be selected cluster-wide using
+`podSelector`. If set with a `namespaceSelector`, all Pods from Namespaces
+selected by the `namespaceSelector` will be selected. Specific Pods from
+specific Namespaces can be selected by providing both a `podSelector` and a
+`namespaceSelector`. Empty `appliedTo` selects nothing. The field is mandatory.
+
+### Direction
+
+The `direction` field specifies the direction of traffic that should be matched.
+It can be `Ingress`, `Egress`, or `Both`.
+
+### Action
+
+The `action` field specifies which action should be taken for the traffic. It
+can be `Mirror` or `Redirect`. For the `Mirror` action, `targetPort` must be
+set to the port to which the traffic will be mirrored. For the `Redirect`
+action, both `targetPort` and `returnPort` need to be specified, the latter of
+which represents the port from which the traffic will be sent back to OVS and be
+forwarded to its original destination.
```

Comment:
This might not be true if the packet has been modified?

Reply:
Ok. Should document the packets should not be modified?

Reply:
Added "Once redirected, a packet should be either dropped or sent back to OVS without modification, otherwise it would lead to undefined behavior."